private/bufpkg/bufmodule/bufmodule.go

```diff
@@ -472,7 +473,36 @@ func ModuleToBucket(
 			return err
 		}
 	}
-	return bufmoduleref.PutDependencyModulePinsToBucket(ctx, writeBucket, module.DependencyModulePins())
+	if err := bufmoduleref.PutDependencyModulePinsToBucket(ctx, writeBucket, module.DependencyModulePins()); err != nil {
+		return err
+	}
+	// This is the default version created by bufconfig getters. The versions should be the
+	// same across lint and breaking configs.
+	version := bufconfig.V1Beta1Version
+	var breakingConfigVersion string
+	if module.BreakingConfig() != nil {
+		breakingConfigVersion = module.BreakingConfig().Version
```

Comment:
Why are we exposing `Version` at all? This feels like an external concern, whereas this should be abstracted away for internal types.

Reply:
This is based because the `Config` structure we store needs to reflect the external type for the digest. The config we store alongside the module needs to reflect changes in the configuration being sent by the user so it can be updated accordingly.